src/init/ClusterManager.ts

```diff
@@ -0,0 +1,114 @@
+import type { Worker } from 'cluster';
+import cluster from 'cluster';
+import { cpus } from 'os';
+import { getLoggerFor } from '../logging/LogUtil';
+
+/**
+ * Different cluster modes.
+ */
+enum ClusterMode {
+  /** Scales in relation to `core_count`. */
+  autoScale = 0,
+  /** Single threaded mode, no clustering */
+  singleThreaded = 1,
+  /** Fixed amount of workers being forked. (limited to core_count) */
+  fixed = 2
+}
+
+/**
+ * Convert workers amount to {@link ClusterMode}
+ * @param workers - Amount of workers (typically passed through cli parameter)
+ * @returns ClusterMode enum value
+ */
+function toClusterMode(workers: number | string): ClusterMode {
+  if (typeof workers === 'string') {
+    workers = Number.parseInt(workers, 10);
+  }
+  // Return enum from 0 to 2 (inclusive)
+  return Math.max(0, Math.min(2, workers));
+}
```

Comment:
I personally am more of a fan of seeing the ClusterMode enum options as opaque objects. So have some if-statements instead of assigning a number to them above and then using that knowledge here.